src/quo2/components/wallet/network_routing/view.cljs

```diff
@@ -0,0 +1,197 @@
+                (gesture/enabled @detecting-gesture?)
+                (gesture/on-begin
+                 (fn [_]
+                   (animation/increase-slider slider-width-shared-value slider-height-shared-value)
+                   (reset! amount-on-gesture-start (reanimated/get-shared-value amount-shared-value))))
+                (gesture/on-update
+                 (fn [event]
+                   (let [new-amount (-> (oops/oget event "translationX")
+                                        (width->amount)
+                                        (+ @amount-on-gesture-start)
+                                        (utils.number/value-in-range 1 max-amount))]
+                     (reanimated/set-shared-value amount-shared-value new-amount))))
+                (gesture/on-finalize
+                 (fn [_]
+                   (animation/decrease-slider slider-width-shared-value slider-height-shared-value)
+                   (animation/hide-slider slider-opacity-shared-value)
+                   (on-new-amount (reanimated/get-shared-value amount-shared-value))
+                   (reset! detecting-gesture? false))))}
+           [:f> slider
+            {:width-shared-value   slider-width-shared-value
+             :height-shared-value  slider-height-shared-value
+             :opacity-shared-value slider-opacity-shared-value}]]]]))))
+
+(defn- add-bar-shared-values
+  [{:keys [amount] :as network}]
+  (assoc network
+         :amount-shared-value      (reanimated/use-shared-value amount)
+         :translate-x-shared-value (reanimated/use-shared-value 0)))
+
+(def ^:private get-negative-amount
+  (comp - reanimated/get-shared-value :amount-shared-value))
+
+(defn network-routing-bars
+  [_]
+  (let [selected-network-idx (reagent/atom nil)
+        selecting-network?   (reagent/atom false)
+        press-locked?        (reagent/atom false)
+        lock-press           #(reset! press-locked? true)
+        unlock-press         #(reset! press-locked? false)
+        hide-bar-animations  (atom {})
+        add-hide-animation   (fn [k anim-props]
+                               (let [new-anim #(animation/hide-pressed-bar anim-props)]
+                                 (swap! hide-bar-animations assoc k new-anim)))
+        reset-state-values   (fn []
+                               (reset! selected-network-idx nil)
+                               (reset! selecting-network? false))]
+    (fn [{:keys [networks total-width total-amount requesting-data? on-amount-selected
+                 add-new-timeout-fn]}]
+      (let [bar-opacity-shared-value (reanimated/use-shared-value 0)
+            network-bars             (map add-bar-shared-values networks)
+            amount->width            #(* % (/ total-width total-amount))
+            bars-widths-negative     (map #(-> % get-negative-amount amount->width)
+                                          network-bars)
+            last-bar-idx             (dec (count network-bars))]
```

Comment:
Well, two things:
1. This component doesn't have designs for empty networks passed, so I think programming for it is making code that will be unused
2. When networks are empty, the component is not rendering bars, so that idx is not a problem:
![photo_2023-10-13_16-19-47](https://github.com/status-im/status-mobile/assets/90291778/95463009-f602-414a-a93f-6f8f9dfdbdef)

Reply:
It's fine as is, but to me the code should be more robust than the designs because designers often don't think about edge cases. The more robust approach is to call `(max 0 (dec (count network-bars)))` to at least fallback to a valid index.